### PR TITLE
Stats revamp v2 detail screens date pickers

### LINKED ISSUE #16684: [android] Stats: Bugs, styles and refining
- [x] Card vertical paddings
- [x] Card radius to 10dp
- [x] Graph marker Comma [bug](https://github.com/wordpress-mobile/WordPress-Android/pull/16327#issuecomment-1110221404) on @irfano's device
- [x] Default cards
- [x] "Highlights" title in Insights tab
- [x] Update Views And Visitors strings, [slack message](p1652189996973789-slack-C02UJ9192TC)
- [x] Add date pickers on new detail screens.
- [x] ~~Remove total cards border in total detail screens.~~
- [x] @irfano: At first opening, some cards are not being shown in detail screens in the emulator. Check if that is reproducible in real devices.
- [x] @irfano: In `StatsFragment` `viewModels()` is being used instead `activityViewModels()`, check if that causes a problem, and fix.
- [x] Views & Visitors crash when talkback is on, LineChartAccessibilityHelper.kt:32 com.github.mikephil.charting.data.Entry cannot be cast to com.github.mikephil.charting.data.BarEntry
- [x] Dates in line chart are "null" in @irfano's emulator.
- [x] Title style, and layout problem 👇

### PR DESCRIPTION
**Pull request by ravishanker**
This PR adds date selector on Insights detail screens

NOTE:
- This PR only adds a date selector feature.  Cards on each detail screen may not be working yet with date selection
- _Each individual detail screen needs a separate PR to make sure the date selection actually reflects in the cards on the respective screen._

Fixes an outstanding item from 16684 

|Detail screen with date selector|
|---|
||

To test:

Setup:
1. Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
2. Select Debug Settings
3. Find StatsRevampV2FeatureConfig under Features in development enable it and restart app
4. Launch / Re-Launch app
5. Go to Stats either using quick links or menu
6. Switch to Insights tab if necessary
7. Make sure Views & Visitors, Total Likes, Total Comments and Total Followers cards are visible on Insights tab. If those cards are not present then add them through stats management using either [+ add new stats card ] at the bottom or clicking ⚙️ icon at the top right hand corner.

Test:
- Click on View More on Views & Visitors, Total Likes, Total Comments and Total Followers cards.
- Ensure it takes to respective detail screen
- Ensure a date selector is visible at the top 
- Make sure the title represents the respective card.  For e.g. Total Likes

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing all the paths

3. What automated tests I added (or what prevented me from doing so)
updated existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by irfano — COMMENTED** [review]
I tested detail screens, and there are some missings in detail screens after the PR.
**Views & Visitors detail:** 
- There should be a pie chart in the referrers card.
- There shouldn't be VIEW MORE button on the Views & Visitors card.

**Total Likes detail:**
- There shouldn't be VIEW MORE button in the Total Likes card.

**Total Comments detail:**
- There should be a total comments card.

**Comment by ravishanker** [comment]
> I tested detail screens, and there are some missings in detail screens after the PR. **Views & Visitors detail:**
> 
>     * There should be a pie chart in the referrers card.
> 
>     * There shouldn't be VIEW MORE button on the Views & Visitors card.
> 
> 
> **Total Likes detail:**
> 
>     * There shouldn't be VIEW MORE button in the Total Likes card.
> 
> 
> **Total Comments detail:**
> 
>     * There should be a total comments card.

That's all correct.  We need to re-look at fixing them in separate small PRs for each detail screen, taking care of the granularity for each card.  The challenge seems to be, unlike existing day/week/month screens where all the cards have same granularity the new insights details screens have different granularity For e.g. days vs weeks.  In fact, we're mixing insights and granular use cases on these details screens!   We need to deal with case by case and card by card to cater for date selection on detail screens

**Comment by irfano** [comment]
> That's all correct. We need to re-look at fixing them in separate small PRs for each detail screen, taking care of the granularity for each card. The challenge seems to be, unlike existing day/week/month screens where all the cards have same granularity the new insights details screens have different granularity For e.g. days vs weeks. In fact, we're mixing insights and granular use cases on these details screens! We need to deal with case by case and card by card to cater for date selection on detail screens

Gotcha! I feel that granularity should be the same in the detail screen. When I see a date picker at the top, I assume all the data is in that date range. Should we discuss this with the team or leave it to the next phase?

**Comment by ravishanker** [comment]
> Gotcha! I feel that granularity should be the same in the detail screen. When I see a date picker at the top, I assume all the data is in that date range. Should we discuss this with the team or leave it to the next phase?

We discussed it here before on channel p1652941827611969/1652442420.245069-slack-C02UJ9192TC.  We need to use WEEK granularity I guess.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt [hunk 1]
```diff
@@ -173,7 +180,10 @@ class ViewsAndVisitorsUseCase
             val selectedDate = dateFromProvider ?: availableDates.last()
             val index = availableDates.indexOf(selectedDate)
 
+            selectedDateProvider.selectDate(selectedDate, availableDates.takeLast(visibleLineCount), statsGranularity)
+
             val selectedItem = domainModel.dates.getOrNull(index) ?: domainModel.dates.last()
+            val previousItem = domainModel.dates.getOrNull(domainModel.dates.indexOf(selectedItem) - 1)
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt:186]: `previousItem` is not being used anywhere.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt [hunk 2]
```diff
@@ -443,19 +444,19 @@ class StatsModule {
...
         return BaseListUseCase(
                 bgDispatcher,
                 mainDispatcher,
                 statsSiteProvider,
-                useCases,
-
+                useCasesFactories.map { it.build(WEEKS, BLOCK) },
                 {
-                    listOf(VIEWS_AND_VISITORS, TimeStatsType.REFERRERS, TimeStatsType.COUNTRIES)
+                    listOf(TimeStatsType.OVERVIEW, TimeStatsType.REFERRERS, TimeStatsType.COUNTRIES)
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt:457]: We used `TimeStatsType.OVERVIEW` in stats types but the first card is not overview actually. Did you plan to update it in the separate PR?

**Review by irfano — APPROVED** [review]
Good work, LGTM!
Just we can remove `previousItem` if it's redundant, then we can merge.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt (hunk 2 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt:457]: It is based on how granular use cases are implemented.  We'll see whether it works or we need a type here when doing detail screens

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt (hunk 1 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/ViewsAndVisitorsUseCase.kt:186]: Probably need it, if not will remove it in separate PR

**Pull request merged**